reactor-netty-http/src/test/java/reactor/netty/http/client/Http2PooledConnectionProviderCustomMetricsTest.java

```diff
@@ -0,0 +1,183 @@
+				.maxConnections(10)
+				.build();
+
+		CountDownLatch latch = new CountDownLatch(5);
+		HttpClient httpClient = createClient(pool, disposableServer::address)
+				.protocol(H2)
+				.secure(spec -> spec.sslContext(sslClient))
+				.doOnConnected(connection -> latch.countDown());
+
+		IntStream.range(0, 5)
+				.forEach(unUsed -> httpClient.get()
+						.uri("/")
+						.responseSingle((resp, bytes) -> bytes.asString())
+						.subscribe()
+				);
+
+		assertThat(latch.await(5, TimeUnit.SECONDS)).isTrue();
+		assertThat(isRegistered.get()).isTrue();
+		assertThat(metrics.get().activeStreamSize()).isEqualTo(5);
+
+		pool.disposeLater().block();
+
+		assertThat(isDeregistered.get()).isTrue();
+	}
+
+	@Test
+	void measurePendingStreamsSize() {
+		AtomicBoolean isRegistered = new AtomicBoolean();
+		AtomicBoolean isDeregistered = new AtomicBoolean();
+		AtomicReference<HttpConnectionPoolMetrics> metrics = new AtomicReference<>();
+
+		disposableServer = createServer()
+				.protocol(H2)
+				.wiretap(true)
+				.secure(spec -> spec.sslContext(sslServer))
+				.handle((req, resp) -> resp.sendString(
+								Mono.delay(Duration.ofSeconds(1))
+										.then(Mono.just("test"))
+						)
+				)
+				.bindNow();
+
+		CustomHttp2MeterRegistrar registrar = new CustomHttp2MeterRegistrar(isRegistered, isDeregistered, metrics);
+		ConnectionProvider pool = ConnectionProvider.builder("custom-pool")
+				.metrics(true, () -> registrar)
+				.maxConnections(1)
+				.pendingAcquireMaxCount(5)
+				.pendingAcquireTimeout(Duration.ofSeconds(20))
+				.build();
+
+		HttpClient httpClient = createClient(pool, disposableServer::address)
+				.protocol(H2)
+				.secure(spec -> spec.sslContext(sslClient))
+				.http2Settings(builder -> {
+					builder.maxStreams(1);
+					builder.maxConcurrentStreams(1);
+				});
+
+		IntStream.range(0, 5)
+				.forEach(unUsed -> {
+					httpClient.get()
+							.uri("/")
+							.responseSingle((resp, bytes) -> bytes.asString())
+							.subscribe();
+				});
+
+		assertThat(isRegistered.get()).isTrue();
+		assertThat(metrics.get().pendingAcquireSize()).isEqualTo(4);
+
+		pool.disposeLater().block();
+
+		assertThat(isDeregistered.get()).isTrue();
+	}
+
+	static final class CustomHttp2MeterRegistrar extends HttpMeterRegistrarAdapter {
+		AtomicBoolean isRegistered;
+		AtomicBoolean isDeregistered;
+		AtomicReference<HttpConnectionPoolMetrics> metrics;
+
+		CustomHttp2MeterRegistrar(
+				AtomicBoolean isRegistered,
+				AtomicBoolean isDeregistered,
+				AtomicReference<HttpConnectionPoolMetrics> metrics
+		) {
+			this.isRegistered = isRegistered;
+			this.isDeregistered = isDeregistered;
+			this.metrics = metrics;
+		}
+
+		@Override
+		public void registerMetrics(String poolName, String id, SocketAddress remoteAddress, HttpConnectionPoolMetrics metrics) {
+			if (isRegistered != null) {
+				isRegistered.set(true);
+			}
+
+			if (this.metrics.get() == null) {
+				this.metrics.set(metrics);
+			}
```

Comment:
Let use `compareAndSet`

```suggestion
			this.metrics.compareAndSet(null, metrics);
```